internal/datacoord/server.go

```diff
@@ -1113,31 +1113,43 @@ func (s *Server) loadCollectionFromRootCoord(ctx context.Context, collectionID i
 	return nil
 }
 
+// update auto balance config loop
 func (s *Server) updateBalanceConfigLoop(ctx context.Context) {
-	log := log.Ctx(s.ctx).WithRateGroup("dc.updateBalanceConfigLoop", 1, 60)
-	ticker := time.NewTicker(Params.DataCoordCfg.CheckAutoBalanceConfigInterval.GetAsDuration(time.Second))
-	for {
-		select {
-		case <-ctx.Done():
-			log.Info("update balance config loop exit!")
-			return
-
-		case <-ticker.C:
-			r := semver.MustParseRange("<2.3.0")
-			sessions, _, err := s.session.GetSessionsWithVersionRange(typeutil.DataNodeRole, r)
-			if err != nil {
-				log.Warn("check data node version occur error on etcd", zap.Error(err))
-				continue
+	// try to update blance config
+	s.updateBalanceConfig()
+
+	if !Params.DataCoordCfg.AutoBalance.GetAsBool() {
+		ticker := time.NewTicker(Params.DataCoordCfg.CheckAutoBalanceConfigInterval.GetAsDuration(time.Second))
+		go func() {
```

Comment:
Should we consider adding a waitgroup to wait for the coroutine to exit gracefully?